src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/RazorFoldingRangeResponse.cs

```diff
@@ -2,9 +2,9 @@
 // Licensed under the MIT license. See License.txt in the project root for license information.
 
 using System.Collections.Generic;
-using OmniSharp.Extensions.LanguageServer.Protocol.Models;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer.Folding
 {
-    internal record RazorFoldingRangeResponse(ICollection<FoldingRange> HtmlRanges, ICollection<FoldingRange> CSharpRanges);
+    internal record RazorFoldingRangeResponse(IEnumerable<FoldingRange> HtmlRanges, IEnumerable<FoldingRange> CSharpRanges);
```

Comment:
nit: Would it be possible to use `IReadOnlyList<FoldingRange>` instead of `IEnumerable` here? That way we could avoid the iterators when iterating the collections

Reply:
Should we just use immutable types here?